lib/loader.js

```diff
@@ -25,6 +31,20 @@ function readFileAsync(filename, encoding) {
     });
 }
 
+const fetchUrl = url => {
+    return fetch(url).then(response => {
```

Comment:
Since you are already using `async/await` you can make this a little more less "thenny"
```
const response = await fetch(url);

if (response.ok) {
    try {
        return await response.json();
    } catch (error) {
        return Promise.reject(new ReadError('Invalid JSON: ' + error.message));
    }

    if (response.status === 404) {
        return Promise.reject(new OpenError('Page not found: ' + url));
    }

    return Promise.reject(new NetworkError('HTTP error: ' + response.status)); 
}

```

I am not entirely sure if returning `Promise.reject` is the best approach or just throwing an error.

Reply:
Happy to help ^_^